sfwrapper/src/ifabricapplicationmanagementclient10sync.cpp

```diff
@@ -3,42 +3,222 @@
+#define FABRIC_DELETE_APPLICATION_DESCRIPTION_VALUES_OR_NULL(desc) \
+    "", \
+    ((desc) == NULL) ? L"NULL" : (desc)->ApplicationName, \
+    MU_BOOL_VALUE(((desc) == NULL) ? 0 : (desc)->ForceDelete), \
+    ((desc) == NULL) ? NULL : (desc)->Reserved
+
+// Format specifiers for FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION
+#define PRI_FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION "s(FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION){ .BuildPath=%ls, .Async=%" PRI_BOOL ", .Reserved=%p }"
+#define FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION_VALUES_OR_NULL(desc) \
+    "", \
+    ((desc) == NULL) ? L"NULL" : (desc)->BuildPath, \
+    MU_BOOL_VALUE(((desc) == NULL) ? 0 : (desc)->Async), \
+    ((desc) == NULL) ? NULL : (desc)->Reserved
+
+// Format specifiers for FABRIC_UNPROVISION_APPLICATION_TYPE_DESCRIPTION
+#define PRI_FABRIC_UNPROVISION_APPLICATION_TYPE_DESCRIPTION "s(FABRIC_UNPROVISION_APPLICATION_TYPE_DESCRIPTION){ .ApplicationTypeName=%ls, .ApplicationTypeVersion=%ls, .Async=%" PRI_BOOL ", .Reserved=%p }"
+#define FABRIC_UNPROVISION_APPLICATION_TYPE_DESCRIPTION_VALUES_OR_NULL(desc) \
+    "", \
+    ((desc) == NULL) ? L"NULL" : (desc)->ApplicationTypeName, \
+    ((desc) == NULL) ? L"NULL" : (desc)->ApplicationTypeVersion, \
+    MU_BOOL_VALUE(((desc) == NULL) ? 0 : (desc)->Async), \
+    ((desc) == NULL) ? NULL : (desc)->Reserved
+
+// Format specifiers for FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION_BASE
+#define PRI_FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION_BASE "s(FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION_BASE){ .Kind=%" PRI_MU_ENUM ", .Value=%p }"
+#define FABRIC_PROVISION_APPLICATION_TYPE_DESCRIPTION_BASE_VALUES_OR_NULL(desc) \
+    "", \
+    MU_ENUM_VALUE(FABRIC_PROVISION_APPLICATION_TYPE_KIND, ((desc) == NULL) ? FABRIC_PROVISION_APPLICATION_TYPE_KIND_INVALID : (desc)->Kind), \
+    ((desc) == NULL) ? NULL : (desc)->Value
+
+
+
 HRESULT FAMC10_ProvisionApplicationType(
     IFabricApplicationManagementClient10 * This,
     /* [in] */ LPCWSTR applicationBuildPath,
     /* [in] */ DWORD timeoutMilliseconds)
 {
-    return ServiceFabric_DoX_NoResult(This, &IFabricApplicationManagementClient10::BeginProvisionApplicationType, &IFabricApplicationManagementClient10::EndProvisionApplicationType,
+    LogInfo("FAMC10_ProvisionApplicationType called with This=%p, applicationBuildPath=%ls, timeoutMilliseconds=%" PRIu32 "",
+        This, applicationBuildPath, timeoutMilliseconds);
+
+    HRESULT result = ServiceFabric_DoX_NoResult(This, &IFabricApplicationManagementClient10::BeginProvisionApplicationType, &IFabricApplicationManagementClient10::EndProvisionApplicationType,
         timeoutMilliseconds,
         applicationBuildPath);
+
+    if (FAILED(result))
+    {
+        LogHRESULTError(result, "FAMC10_ProvisionApplicationType returning");
+    }
+    else
+    {
+        LogHRESULTInfo(result, "FAMC10_ProvisionApplicationType returning");
```

Comment:
>LogHRESULTInfo [](http://example.com/codeflow?start=8&length=14)

Verbose also. Likely the bot will help change them to verbose in a very bulk way.